src/CookiePhysicsComponent.cpp

```diff
@@ -73,11 +74,45 @@ void CookiePhysicsComponent::updatePhysics(float deltaTime) {
         // The color change has to be done here,
         // due to the bouncing off, the cookie never really hits the object.
         // TODO(nurgan) implement checking for object type and only change color if it is a "house"
-        if(objHit->getRenderComponent()->getShader().compare("Green")) {
+        if( objHit->getRenderComponent()->getShader().compare("Green")) {
             objHit->changeShader("Green");
+            cookieState.hits++;
```

Comment:
Yes, a house needs to know if its already been hit. By now I just used the check which Shader is used if its already been hit. I'll add a actual "already hit status"